splinterd/src/main.rs

```diff
@@ -443,25 +446,52 @@ fn main() {
 
     let matches = app.get_matches();
 
-    let log_level = match matches.occurrences_of("verbose") {
-        0 => log::LevelFilter::Warn,
-        1 => log::LevelFilter::Info,
-        2 => log::LevelFilter::Debug,
-        _ => log::LevelFilter::Trace,
-    };
-
-    let mut log_spec_builder = LogSpecBuilder::new();
-    log_spec_builder.default(log_level);
-    log_spec_builder.module("hyper", log::LevelFilter::Warn);
-    log_spec_builder.module("tokio", log::LevelFilter::Warn);
-    #[cfg(feature = "https-bind")]
-    log_spec_builder.module("h2", log::LevelFilter::Warn);
-
-    Logger::with(log_spec_builder.build())
-        .format(log_format)
-        .log_target(flexi_logger::LogTarget::StdOut)
-        .start()
-        .expect("Failed to create logger");
+    #[cfg(feature = "log-config")]
+    {
+        let config_file = get_config_file(&matches);
+        let config_file_path = if Path::new(config_file).is_file() {
+            Some(config_file)
+        } else {
+            None
+        };
+
+        let _log_handle = match create_config(config_file_path, matches.clone()) {
+            Ok(config) => {
+                if let Some(log_config) = config.log_config() {
+                    Some(
+                        log4rs::init_raw_config(log_config.clone())
+                            .expect("Failed to create logger"),
+                    )
+                } else {
+                    None
+                }
+            }
+            Err(e) => {
+                println!("Could not create config: {}", e);
```

Comment:
Errors must be printed with eprintln! and should begin with "Error: ".

Reply:
Changed to an expect to keep with the current behavior of stopping the app if configs are broken. Maybe it makes sense to start the app without logging but don't think its a good idea